### PR TITLE
Quark AJP10304 layout

### PR DESCRIPTION
**Pull request by ajp10304**
Support AJP10304 keymap / userspace for Quark.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/quark/keymaps/ajp10304/keymap.c [hunk 1]
```diff
@@ -0,0 +1,185 @@
+#include "ajp10304.h"
```
Comment (drashna) [inline keyboards/quark/keymaps/ajp10304/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/quark/keymaps/ajp10304/keymap.c (hunk 1 above)
Comment (ajp10304) [reply keyboards/quark/keymaps/ajp10304/keymap.c:17]: Done.

Comment (ajp10304) [reply keyboards/quark/keymaps/ajp10304/keymap.c:17]: Updated.

File: keyboards/quark/keymaps/ajp10304/rules.mk [hunk 2]
```diff
@@ -0,0 +1 @@
+MOUSEKEY_ENABLE = yes
```
Comment (ajp10304) [inline keyboards/quark/keymaps/ajp10304/rules.mk:1]: Adding `DYNAMIC_MACRO_ENABLE = yes` in here means the board stops working completely.  It works in my other keymaps.  Any idea why the Quark would be different?

Comment (drashna) [reply keyboards/quark/keymaps/ajp10304/rules.mk:1]: Dynamic macros uses a lot of memory.  My guess is that the chip doesn't have enough for everything enabled.

**Review by drashna — APPROVED** [review]

File: keyboards/quark/keymaps/ajp10304/rules.mk (hunk 2 above)
Comment (ajp10304) [reply keyboards/quark/keymaps/ajp10304/rules.mk:1]: Hmm, I did wonder about that. Thanks for the reply.

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**